src/build_system.ml

```diff
@@ -1528,7 +1550,7 @@ end = struct
   let rules_for_files rules deps =
     Dep.Set.paths deps ~eval_pred
     |> Path.Set.fold ~init:Rule.Set.empty ~f:(fun path acc ->
-      match Path.Map.find rules path with
+      match Path.Build.Map.find rules (Path.as_in_build_dir_exn path) with
```

Comment:
Maybe this shouldn't raise? I'm not sure how exactly `request` is parsed, but it seems OK (even though redundant) for the user to say that they want a source file to be up to date.